user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -1927,6 +1933,10 @@ define([
     return response;
   }
 
+  function _isRepositoryPath(path) {
+    return path.charAt(0) === "/" || path.charAt(0) === ":";
+  };
```

Comment:
No need for semi-colon after `}`.
Also, given this is not a method after all, you could drop the underscore `_` prefix, which is used to distinguish private/protected methods.

Reply:
I see underscores here and there and I saw one in your comment [here](https://github.com/pentaho/pentaho-platform/pull/5562#discussion_r1533763969)

I will remove it. Do you know why they are in the file at all? js doesn't have methods at all, right?

Reply:
When I remove it I see the following error:

```browser.js:361 Uncaught ReferenceError: Cannot access 'isRepositoryPath' before initialization
    at child.updateFolderButtons (browser.js:361:32)
    at child.updateFolderClicked (browser.js:357:12)
    at triggerEvents (backbone.js:230:51)
    at child.trigger (backbone.js:163:19)
    at child.set (backbone.js:387:16)
    at child.clickFolder (browser.js:1308:18)
    at HTMLDivElement.dispatch (jquery.js?context=mantle&cssOnly=true:5429:27)
    at elemData.handle (jquery.js?context=mantle&cssOnly=true:5233:28)

Reply:
The underscores are just a convention to denote private/protected. They're mostly used for methods of classes. This function is an _inner_ function --- meaning it's declared nested inside another function, and is thus only accessible from inside its "parent" function. It can be accessed from anywhere inside the parent function, either before or after its declaration, given declarations are always implicitly "hoisted" to the top of the parent function by the JS interpreter. This function is not a method of any class or object in the file (would have to be the value of a property in the prototype/instance object of a constructor function to be one), and so this convention does not so needed in this case, given visibility of the function is already controlled and limited to the declaration's containing scope.

For completeness, still, sometimes, you'll see the underscore being used for these functions, to denote they're declared somewhere in the file, in one of the containing function scopes, but not the immediate one. This makes these distinguishable from module imports (requested using the RequireJS system), which are inherently public values... This convention is less common.

Reply:
ooooohhhhhh. I think I know what is going on. In removing the _ from `_isRepositoryPath` I have the statement `const isRepositoryPath = isRepositoryPath(_folderPath);` and since a function can be a variable, I am creating a broken statement. I will rename the variables...